raft/raft.go

```diff
@@ -654,6 +671,11 @@ func stepLeader(r *raft, m pb.Message) {
 					// an update before, send it now.
 					r.sendAppend(m.From)
 				}
+				// Transfer leadership is in progress.
+				if m.From == r.leadTransferee && pr.Match == r.raftLog.lastIndex() && !r.timeoutNowSent {
+					r.logger.Infof("%x sent MsgTimeoutNow to %x after received MsgAppResp", r.id, m.From)
```

Comment:
Yeah, this optimization can wait. One thing we could do to avoid the scenario in which the follower rejects the MsgApp and then starts an election that fails is to include the leader's last log index in the MsgTimeoutNow. The follower would only start campaigning if it matches the index; if it doesn't then it knows it can't win.